File: tests/visualtests/test_neighborhood.cpp
#include "engine.cpp"
#include "readmesh.cpp"

int main()
{
    char fname[] = "../data/test_mesh2d.off";
    int nv, nt;
    double *ver, *orient;
    int *tri;
    read_off(fname, nv, nt, ver, tri);
    dummy_orient(nt, orient);
    TriangularMesh S;
    S.initialize(nv, nt, ver, tri, orient, 0, 0);
    double radius = 2.0;
    S.compute_neighborhood(radius, 0);

    // check t_neigh_tri
    FILE* file = fopen("test_mesh2d.neightri", "w");
    fprintf(file, "%.5f\n", radius);
    for (int i=0;i<S.nt;i++) {
        for (int j=0;j<S.t_nb_extneigh_tri[i];j++)
            fprintf(file, "%d ", S.t_extneigh_tri[i][j]);
        fprintf(file, "\n");
    }
    fclose(file);
}

File: tests/visualtests/test_neighborhood.py
import os
import numpy as np
import matplotlib.pyplot as plt
from readmesh import read_off


ver, tri = read_off('../data/test_mesh2d.off')

with open('test_mesh2d.neightri', 'rt') as file:
    radius = float(file.readline())
    lines = file.readlines()
os.remove('test_mesh2d.neightri')
os.remove('test')

theta = np.linspace(0, 2*np.pi, 200)
cx = radius * np.cos(theta)
cy = radius * np.sin(theta)
i = 0
for line in lines:
    I = np.array([int(s) for s in line.split()])
    plt.triplot(ver[:, 0], ver[:, 1], tri, c=(0.8,)*3)
    plt.triplot(ver[:, 0], ver[:, 1], tri[I], c='r')
    for k in range(3):
        G = ver[tri[i, k]]
        plt.plot(G[0] + cx, G[1] + cy, 'C0')
    plt.axis('equal')
    plt.show()
    i += 1

